src/config/navigation/config.ts

```diff
@@ -2,51 +2,62 @@ import * as Linking from 'expo-linking';
 
 import { NavigatorConfig, ScreenName } from '../../types';
 
-// import { tabNavigatorConfig } from './tabConfig';
-
-// export const navigatorConfig: NavigatorConfig = {
-//   type: 'tab',
-//   config: tabNavigatorConfig
-// };
-
-// const index = 2;
-
-// export const linkingConfig = {
-//   prefixes: [Linking.createURL('/')],
-//   config: {
-//     screens: {
-//       // For tab navigation choose the preferred tab by its position in the config array
-//       [`Stack${index}`]: {
-//         // The initialRouteName has to be the initial route of the chosen stack
-//         // (Home for drawer navigation, and whatever is specified in the tab config for tab navigation)
-//         initialRouteName: navigatorConfig.config.tabConfigs[index].stackConfig.initialRouteName,
-//         screens: {
-//           [ScreenName.EncounterUserDetail]: 'encounter',
-//           [ScreenName.Home]: '*'
-//         }
-//       }
-//     }
-//   }
-// };
-
-export const navigatorConfig: NavigatorConfig = {
-  type: 'drawer'
-};
-
-export const linkingConfig = {
-  prefixes: [Linking.createURL('/')],
-  config: {
-    screens: {
-      // For tab navigation choose the preferred tab by its position in the config array: AppStack -> `Stack${index}`
-      AppStack: {
-        // The initialRouteName has to be the initial route of the chosen stack
-        // (Home for drawer navigation, and whatever is specified in the tab config for tab navigation)
-        initialRouteName: ScreenName.Home,
-        screens: {
-          [ScreenName.EncounterUserDetail]: 'encounter',
-          [ScreenName.Home]: '*'
+import { tabNavigatorConfig } from './tabConfig';
+
+import { useSettingsContext } from '../../SettingsProvider';
+
+const settings = useSettingsContext();
+
+let navigatorConfig: NavigatorConfig;
+let linkingConfig: any;
+
+if (settings.globalSettings.navigation != "drawer") {
+  const index = 2;
```

Comment:
Which is more useful and more compatible for the whole code, this code or the one I shared below?

```
import * as Linking from 'expo-linking';
import { useContext } from 'react';

import { SettingsContext } from '../../SettingsProvider';
import { NavigatorConfig, ScreenName } from '../../types';

import { tabNavigatorConfig } from './tabConfig';

const { globalSettings } = useContext(SettingsContext);

let navigatorConfig: NavigatorConfig;
let linkingConfig: any;

if (globalSettings.navigation != 'drawer') {
...
```

Reply:
I would say the one you shared below, since its already common practice in the project

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/3bab9a27dd392bcd379a6b71b9663c1b7ca5e805